### PR TITLE
[Angular-mocks] Add optional parameter to IHttpBackendService.verifyNoOutstandingExpectation()

### PR DESCRIPTION
**Pull request by hamxiaoz**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Angular source code for the method](https://github.com/angular/angular.js/blob/master/src/ngMock/angular-mocks.js#L1860)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by mhegazy** [comment]
@ersimont abd @mikehaas763  can you please review this change.

**Comment by mikehaas763** [comment]
@hamxiaoz In the angular docs for this function (looking at master docs), I don't see anything mentioned that the function now accepts a boolean which triggers a digest. Is this extremely new and coming in a future angular version?

docs I mentioned above: https://docs.angularjs.org/api/ngMock/service/$httpBackend

**Comment by hamxiaoz** [comment]
@mikehaas763 In the PR I have the source code linked: 
https://github.com/angular/angular.js/blob/master/src/ngMock/angular-mocks.js#L1860

As you can see the parameter is there. 

My coworker pointed out that actually this parameter is added by this PR:
https://github.com/katemihalikova/angular.js/commit/acb462aefe1a55d1eb36a129fc62c3aeb11b9259
Which is solving this issue in Angular.js: https://github.com/angular/angular.js/issues/13506
It seems the PR **forgot** to update the JSDoc comment. That's why it's not displaying in the doc.

I'll create a PR to angualr.js to add the jsdoc comment.

**Comment by ersimont** [comment]
That last link from `hamxiaoz` is the key, in my opinion. I don't know if there is a standard in place for this sort of thing already, but if not I would suggest that purposely undocumented features should not make their way into DefinitelyTyped.

**Comment by mikehaas763** [comment]
@hamxiaoz sorry, I missed your original link. Thanks for the other links though.

Sorry @ersimont, your point is reasonable, but hear me out. Type definitions should represent actual code, not the documentation, of public APIs. It's been added there so it can be called from JavaScript, despite it possibly allowing the consumer to shoot themselves in the foot. Therefore, we should also be able to call it from TypeScript. What do you think?

**Comment by ersimont** [comment]
In principle, "purposely undocumented" sounds like "we would make this private (or protected or package) if Javascript had such a feature." Or "this is not meant to be used by the public". So I still think it does not belong in DefinitelyTyped. If people want to use it in typescript they would have to do something like this, which is ugly but captures the idea "I know I'm using this in a way that was not intended": 

```typescript
($httpBackendService.verifyNoOutstandingExpectation as Function)(false);
```

And to be clear: I'm speaking of **purposely** undocumented code. I.e. code that the author(s) specifically chose not to document because they don't think people should be using it / know about it.  If it is something the docs simply missed because they are incomplete, that would be a different issue.

I'll leave a couple comments on the code for the case that this is merged in spite of my protest :)

File: angular-mocks/index.d.ts [hunk 1]
```diff
@@ -132,8 +132,9 @@ declare module 'angular' {
 
     /**
       * Verifies that all of the requests defined via the expect api were made. If any of the requests were not made, verifyNoOutstandingExpectation throws an exception.
+      * @param digest Do digest before checking expectation. Pass anything expect false to trigger digest.
```
Comment (ersimont) [inline angular-mocks/index.d.ts]: If this is merged in spite of my protest:
* Can we indicate here that this is a "private" feature which is not actually meant to be used in normal client code? If angular purposely chose to leave it undocumented, that is good information for users to know.
* Typo: "Pass anything ~expect~ **except** false..."

**Comment by mikehaas763** [comment]
Is it accurate to say that it "is not actually meant to be used"? Why would it have been added? It wasn't added for internal usage of the lib was it? To my knowledge it was added for consumers to use. Them not documenting it so people don't shoot themselves in the foot is orthogonal to it being a public API.

**Comment by hamxiaoz** [comment]
I have updated the comment, ready to merge :)

**Pull request merged**